basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -0,0 +1,880 @@
+                size_t min_idx = std::get<1>(min_pair);
+                if (min_idx == -1)
+                {
+                    break;
+                }
+
+                visited_idx.insert(min_idx); // Mark this point as visited
+
+                double v_i = std::get<0>(min_pair);
+                double x_i = downtracks[min_idx];
+                for (int i = min_idx - 1; i > 0; i--)
+                {   // NOTE: Do not use size_t for i type here as -- with > 0 will result in overflow
+                    //       First point's speed is left unchanged as it is current speed of the vehicle
+                    double v_f = curv_speeds[i];
+                    double dv = v_f - v_i;
+
+                    double x_f = downtracks[i];
+                    double dx = x_f - x_i;
+
+                    if (dv > 0)
+                    {
+                        v_f = std::min(v_f, sqrt(v_i * v_i - 2 * accel_limit * dx)); // inverting accel as we are only visiting deceleration case
+                        visited_idx.insert(i);
+                    }
+                    else if (dv < 0)
+                    {
+                        break;
+                    }
+                    output[i] = v_f;
+                    v_i = v_f;
+                    x_i = x_f;
+                }
+            }
+
+            log::printDoublesPerLineWithPrefix("only_reverse[i]: ", output);
+
+            output = trajectory_utils::apply_accel_limits_by_distance(downtracks, output, accel_limit, accel_limit);
+            log::printDoublesPerLineWithPrefix("after_forward[i]: ", output);
+
+            return output;
+        }
+
+        std::vector<cav_msgs::TrajectoryPlanPoint> trajectory_from_points_times_orientations(
+            const std::vector<lanelet::BasicPoint2d> &points, const std::vector<double> &times, const std::vector<double> &yaws,
+            ros::Time startTime)
+        {
+            if (points.size() != times.size() || points.size() != yaws.size())
+            {
+                throw std::invalid_argument("All input vectors must have the same size");
+            }
+
+            std::vector<cav_msgs::TrajectoryPlanPoint> traj;
+            traj.reserve(points.size());
+
+            for (int i = 0; i < points.size(); i++)
+            {
+                cav_msgs::TrajectoryPlanPoint tpp;
+                ros::Duration relative_time(times[i]);
+                tpp.target_time = startTime + relative_time;
+                tpp.x = points[i].x();
+                tpp.y = points[i].y();
+                tpp.yaw = yaws[i];
+
+                tpp.controller_plugin_name = "default";
+                //tpp.planner_plugin_name        //Planner plugin name is filled in the tactical plugin
+
+                traj.push_back(tpp);
+            }
+
+            return traj;
+        }
+
+
+        std::vector<PointSpeedPair> attach_back_points(const std::vector<PointSpeedPair> &points,
```

Comment:
Doesn't have to change, but "attach_back_points" is also kind of a weak name. What are "back_points" and why do I need them?